imagekit-sdk/src/main/java/io/imagekit/sdk/tasks/UrlGen.java

```diff
@@ -58,13 +66,21 @@ private static String generateUrl(
                 newUrl=buildFullUrl(urlEndpoint+(path.charAt(0)=='/'?path.substring(1,path.length()):path),queryParameters,transformation,transformationPosition,signed,expireSeconds, config.getPrivateKey(), urlEndpoint);
             }
             else {
-                newUrl = buildPathUrl(path.charAt(0)=='/'?path:"/"+path, urlEndpoint, queryParameters, transformation, transformationPosition, signed, expireSeconds, config.getPrivateKey());
+                newUrl = buildPathUrl(path, urlEndpoint, queryParameters, transformation, transformationPosition, signed, expireSeconds, config.getPrivateKey());
             }
         }
         return newUrl;
     }
 
-    private static String buildPathUrl(String path, String urlEndpoint, Map<String, String> queryParameters, List<Map<String, String>> transformation, String transformationPosition, boolean signed, long expireSeconds, String privateKey) {
+    private static String buildPathUrl(String path, String urlEndpoint, Map<String, String> queryParameters, List<Map<String, String>> transformation, String transformationPosition, boolean signed, long expireSeconds, String privateKey) throws Exception{
+        // First strip off any leading /
+        path = path.startsWith("/") ? path.substring(1) : path;
+
+        // Then ensure the supplied path is always url encoded
+        path = URLEncoder.encode(path, StandardCharsets.UTF_8.name());
+
+        // Then ensure it always begins with a /
+        path = path.charAt(0) == '/' ? path : "/" + path;
```

Comment:
Why would it still have a '/' at the start? Is this actually possible?

Reply:
This is the exact code they had before. Don't want to touch it. Used to be done by the caller.
👎